virtualenv/builders/legacy.py

```diff
@@ -175,54 +208,94 @@ def create_virtual_environment(self, destination):
-        # only have the os module but none of the os's modules dependencies or
-        # any other module unless they are "special" modules built into the
-        # interpreter like the sys module.
-        copyfile(
-            os.path.join(base_python["lib"], "os.py"),
-            os.path.join(lib_dir, "os.py"),
-        )
-
         # The site module has a number of required modules that it needs in
         # order to be successfully imported, so we'll copy each of those module
         # into our virtual environment's lib directory as well. Note that this
         # list also includes the os module, but since we've already copied
         # that we'll go ahead and omit it.
+        sys_prefix = self._python_info["sys.prefix"]
+        lib_dirs = [
+            path for path in self._python_info["sys.path"]
+            if path.startswith(sys_prefix)
+            # TODO: this has an unhandled edgecase, it handle case with
+            # partial match (should only match full components)
+
+        ]
+
+        for module in self.flavor.bootstrap_modules(self._python_info):
+            modulepath = self._locate_module(module, lib_dirs)
+            if modulepath:
+                copyfile(
+                    modulepath,
+                    os.path.join(
+                        destination,
+                        os.path.relpath(modulepath, sys_prefix)
+                    ),
+                )
 
-        for module in self.flavor.core_modules:
-            copyfile(
-                os.path.join(base_python["lib"], module),
-                os.path.join(lib_dir, module),
-            )
+        osmodulepath = self._locate_module("os", lib_dirs)
+        if not osmodulepath:
+            raise RuntimeError("Can't locate os module in any of %s." % lib_dirs)
+        osmoduledestination = os.path.join(
+            destination,
+            os.path.relpath(osmodulepath, sys_prefix)
+        )
+        copyfile(osmodulepath, osmoduledestination)
 
-        dst = os.path.join(lib_dir, "site.py")
-        with io.open(dst, "w") as dst_fp:
+        include_dir = self.flavor.include_dir(self._python_info)
+        copyfile(
+            os.path.join(self._python_info["sys.prefix"], include_dir),
+            os.path.join(destination, include_dir)
+        )
+        copyfile(
+            os.path.join(self._python_info["sys.prefix"], include_dir),
+            os.path.join(destination, "local", include_dir)
+        )
+
+        dst = os.path.join(os.path.dirname(osmoduledestination), "site.py")
+        logger.debug("Writing %s", dst)
+        with io.open(dst, "wb") as dst_fp:
             # Get the data from our source file, and replace our special
             # variables with the computed data.
             data = SITE
-            data = data.replace("__PREFIX__", repr(destination))
-            data = data.replace("__EXEC_PREFIX__", repr(destination))
+            data = data.replace(b"__PREFIX__", self._path_repr(destination))
```

Comment:
If anyone is wondering about the weird `_path_repr` it's just my poor attempt to deal with various string/bytes literals over all the pythons. I still need to add integration tests with weird unicode chars in them.